aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,316 @@
+        error_list.append('Wrong file version: ' + nc_file_version)
+
+    for variable in required_variables:
+        if variable not in file_variables:
+            error_list.append(variable + ' variable missing')
+        else:
+            VoIdimensions = list(nc[variable].dims)
+            if 'TIME' not in VoIdimensions:
+                error_list.append('TIME is not a dimension for ' + variable)
+            if 'LATITUDE' in VoIdimensions and len(nc.LATITUDE) > 1:
+                error_list.append('more than one LATITUDE for ' + variable)
+            if 'LONGITUDE' in VoIdimensions and len(nc.LONGITUDE) > 1:
+                error_list.append('more than one LONGITUDE for ' + variable)
+            for dim in VoIdimensions:
+                if dim not in allowed_dimensions:
+                    error_list.append('not allowed dimension: ' + dim)
+
+    if 'NOMINAL_DEPTH' not in file_variables and 'instrument_nominal_depth' not in attributes:
+        error_list.append('no NOMINAL_DEPTH')
+
+    return error_list
+
+
+def get_nvalues(nc):
+    """
+    Get the number of cells above the sensor
+    :param nc: xarray dataset
+    :return: number of cells, number of bins
+    """
+    if 'HEIGHT_ABOVE_SENSOR' in nc.dims:
+        nbins = nc.dims['HEIGHT_ABOVE_SENSOR']
+        nvalues = nc.dims['TIME'] * nbins
+    else:
+        nbins = 1
+        nvalues = nc.dims['TIME']
+    return nvalues, nbins
+
+
+def get_varvalues(nc, varname):
+    """
+    Return a 1D array of 2D values
+    :param nc: dataset
+    :param varname: Variable of interest
+    :return: variable values flattened
+    """
+    return nc[varname].values.flatten()
+
+
+def get_instrumentID(nc):
+    """
+    Create instrument id based on deployment metadata
+    :param nc: xarray dataset
+    :return: instrumentID as string
+    """
+    return '; '.join([nc.deployment_code, nc.instrument, nc.instrument_serial_number])
+
+
+def in_water(nc):
+    """
+    cut data to in-water only timestamps, dropping the out-of-water records.
+    :param nc: xarray dataset
+    :return: xarray dataset
+    """
+    time_deployment_start = np.datetime64(nc.attrs['time_deployment_start'][:-1])
+    time_deployment_end = np.datetime64(nc.attrs['time_deployment_end'][:-1])
+    TIME = nc['TIME'][:]
+    return nc.where((TIME >= time_deployment_start) & (TIME <= time_deployment_end), drop=True)
+
+
+
+
+## MAIN FUNCTION
+def aggregate_velocity(files_to_agg, site_code, base_path):
+    """
+    Aggregate U, V and W CUR variables from all deployments at one site.
+    the vertical cells are flattened and related to its depth
+    additional metadata variables are stored to track the origin of the data
+    :param files_to_agg: list of files to aggregate
+    :param site_code: site code
+    :param base_path: base path to store the resulting file
+    :return: name of the resulting file, list of rejected files
+    """
+
+    varlist = ['UCUR', 'VCUR', 'WCUR', 'DEPTH']
```

Comment:
This block of vars can be defined (promoted) to keyword arguments and be dynamically assigned, if required.

```
agg_options = {'varlist': ..., 'time_units': , ...] # or even a global dict
aggregate_velocity(...,varlist=[...],time_units=...). # or **agg_options
```

Reply:
could be but we don't expect to aggregate anything else apart from those variables.

Reply:
anyway, this is a good practice both codewise and intent wise. For example, when reading, why the function "aggregate velocity" does not provide the option to select which velocity/vars I want to aggregate? 

Code wise, "default" options should hardly be defined in the scope of the function. Also, if you put as kwargs and need to change the parameters you just change the function call instead of changing the function code!